thunder/benchmarks/benchmark_litgpt.py

```diff
@@ -188,6 +199,17 @@ def __init__(
         self.model = self.init_model()
         print(f"Time to instantiate model: {time.perf_counter() - t0:.02f} seconds.")
 
+        # fp8-delayed-te precision for thunder is handled through compile
+        if is_transformer_engine(low_precision_mode) and "thunder" not in self.compile:
+            te_precision = TransformerEnginePrecision(weights_dtype=torch.bfloat16, replace_layers=True)
+            converted_model = te_precision.convert_module(self.model)
+
+            def call_model_with_fp8autocast(*args, **kwargs):
+                with te.fp8_autocast(enabled=True):
+                    return converted_model(*args, **kwargs)
+
+            self.model.__call__ = call_model_with_fp8autocast
```

Comment:
Is this safe?

Reply:
The safer, or "less hacky" way would be to create an appropriate context manager depending on the condition:
```
if is_transformer_engine(low_precision_mode) and "thunder" not in self.compile
```
so it would return either null context, or `fp8_autocast`. So we could have a function returning a proper context manager object, and then use it in each model call. 
@kshitij12345 Or maybe you see a better way? I very liked your suggestion, but I don't know how to make it work without this `self.model.__call__ = call_model_with_fp8autocast` line 🙈

Reply:
I think I would feel much more at ease with the less hacky way of using a context manager, but clever seems OK, just overriding `self.model.__call__` looks risky (after staring down way to many Python corner cases).

Reply:
Ouch, apologies for proposing a risky solution. Let's stick to using appropriate ctx managers.

Reply:
@t-vi @kshitij12345 I added the changes for context managers, but I think the code will change according to the [recent comment](https://github.com/Lightning-AI/lightning-thunder/pull/689#discussion_r1665365705)